### PR TITLE
add electron builds to ci (continued)

### PR DESCRIPTION
**Pull request by kewde**
See #811 for initial PR.

- Fixed GYP_ARGS issue
https://github.com/mapbox/node-sqlite3/pull/976/files#diff-d3bcfe1f6e2dbb503917dadba6a302fcR13
- Fixed merge conflicts

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -28,6 +28,31 @@ environment:
       platform: x64
     - nodejs_version: 9
       platform: x86
+    # electron
+    - nodejs_version: 6
+      platform: x64
+      msvs_toolset: 12
```
Comment (kewde) [inline appveyor.yml]: Is `msvs_toolset` still required?

File: scripts/build_against_electron.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+set -e -u
+
+export DISPLAY=":99.0"
+GYP_ARGS="--runtime=electron --target=${ELECTRON_VERSION} --dist-url=https://atom.io/download/electron"
+NPM_BIN_DIR="$(npm bin -g 2>/dev/null)"
+
+function publish() {
+    if [[ ${PUBLISHABLE:-false} == true ]] && [[ ${COMMIT_MESSAGE} =~ "[publish binary]" ]]; then
+        node-pre-gyp package $GYP_ARGS
```
Comment (kewde) [inline scripts/build_against_electron.sh]: GYP_ARGS fixed

**Comment by kewde** [comment]
@springmeyer please consider this PR.
The scripts are quite straightforward, they merely modify the arguments passed to node-gyp.

The binaries provided by sqlite-offline by Dennis are _not_ suited for all platforms (Debian 8 for example), as they require a GLIBCXX >= 3.4.21. The Ubuntu Trusty environment by Travis will do a better job.

**Comment by springmeyer** [comment]
@kewde. Thanks for working on this! Can you provide me detailed instructions on how to test things are working?  I’m not an electron user so consider me a noob when writing your instructions. In other words, please provide details on how to install electron and how to test and app using node-sqlite3. If I can get things working with your instructions then I’ll be more confortable merging this (and supporting it in the long term).

**Comment by springmeyer** [comment]
@kewde. Also you may be interested in https://github.com/mapbox/node-pre-gyp/pull/343. I presume that will be needed for making this work easy to use. If you agree feel free to create a new PR over at node-pre-gyp that updates that to latest master to be available to test here.

**Comment by kewde** [comment]
@springmeyer 

The initial author was kind enough to update the test suite in this repository to run in the electron environment (for both appveyor and Travis) and it passes successfully.

Appveyor:
https://github.com/mapbox/node-sqlite3/pull/976/files#diff-a1df4cde076721bbb38eb706ab0e7e30R168

Travis:
https://github.com/mapbox/node-sqlite3/pull/976/files#diff-d3bcfe1f6e2dbb503917dadba6a302fcR28

If you truly desire a simple skeleton application for sqlite3 then I will be more than happy to provide one.

**Comment by kewde** [comment]
@springmeyer 
Here you go, a working example (packaging to .exe, .deb, etc included!)
https://github.com/kewde/node-sqlite3-electron-sample

**Comment by kewde** [comment]
The appveyor test suite seems to be failing randomly, more specifically the test  "retrieve 100,000 rows with Statement#each". The last commit 7c765de has a green checkmark - tests ran fine with electron but I doubt it will remain this way.

**Comment by kewde** [comment]
Fixed the Travis coverage test in 534bc4b (note: moved it to linux though)

**Comment by springmeyer** [comment]
Awesome thank you @kewde!!

**Comment by kewde** [comment]
@springmeyer The results for the CI's will take a while (as there are quite a few builds in the backlog) but it should be all green checkmarks. I'm going to grab some sleep, it's 4 AM here.

**Comment by kewde** [comment]
@springmeyer I truly hope this gets landed this week :sweat_smile: , I rely on the sqlite3 dependency and I've had to hack around it in an ugly way.

**Comment by bkniffler** [comment]
Hey, can we get this merged? Electron users would be very happy!

**Comment by kewde** [comment]
Need a way to update libc6 for 2.0.1, currently has 2.15 and errors with: version 'GLIBC_2.17' not found

**Comment by kewde** [comment]
Seems like precise is limited to 2.15 - so I'm giving trust a try.
This will however most likely cause a higher version to be required by node-sqlite3 :/

**Comment by kewde** [comment]
@springmeyer ping

File: scripts/build_against_electron.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+function electron_test() {
+    "$NPM_BIN_DIR"/electron test/support/createdb-electron.js
+    "$NPM_BIN_DIR"/electron-mocha -R spec --timeout 480000
+}
+
+# test installing from source
+npm install --build-from-source  --clang=1 $GYP_ARGS
+
+# TODO: remove me -start
```
Comment (kewde) [inline scripts/build_against_electron.sh]: remove this tho - I've been leeching off your CI to make my own node-sqlite3 bins.

File: appveyor.yml [hunk 4]
```diff
@@ -37,3 +92,6 @@ install:
 build: OFF
 test: OFF
 deploy: OFF
+
+artifacts:
```
Comment (kewde) [inline appveyor.yml]: perhaps remove these too - stores artifacts for windows builds.

**Comment by springmeyer** [comment]
> @springmeyer ping

Thanks for the ping @kewde - I'm back from being offline for several weeks. I plan to take a close look and merge this soon - ideally by early next week. Thanks!

**Comment by kewde** [comment]
I've been using the binaries provided by your CI for well over a month now - on all platforms.

Operating systems:
* win32 ia32
* win32 x64
* linux x64
* macOS

Electron versions:
* v1.7
* v1.8
*v2.0.1

Our releases get well over 3500 users - and when shit breaks - they complain :P 
I just released the first version - using node-sqlite3 bins here.
We've had well over 200 testers in the past and we've never had an issue with the binaries.
You can track our downloads here:
http://www.somsubhra.com/github-release-stats/?username=particl&repository=particl-desktop

**Comment by bhj** [comment]
@kewde Thanks for all your work. Does this handle https://github.com/mapbox/node-pre-gyp/pull/343 as well?

**Comment by kewde** [comment]
ping @springmeyer

**Pull request merged**